src/charm.py

```diff
@@ -233,6 +235,48 @@ def _on_restart_action(self, event):
         container.restart(self.name)
         self.set_active_unit_status()
 
+    @log_event_handler(logger)
+    def _on_update_status(self, event):
+        """Handle `update-status` events.
+
+        Args:
+            event: The `update-status` event triggered at intervals.
+        """
+        try:
+            self._validate()
+        except ValueError:
+            return
+
+        container = self.unit.get_container(self.name)
+        valid_pebble_plan = self._validate_pebble_plan(container)
+        if not valid_pebble_plan:
+            self._update(event)
+            return
+
+        check = container.get_check("up")
+        if check.status != CheckStatus.UP:
+            self.unit.status = MaintenanceStatus("Status check: DOWN")
+            return
+
+        self.set_active_unit_status()
+        if self.unit.is_leader():
+            self.ui._provide_server_status()
+
+    def _validate_pebble_plan(self, container):
+        """Validate Temporal server pebble plan.
+
+        Args:
+            container: application container
+
+        Returns:
+            bool of pebble plan validity
+        """
+        try:
+            plan = container.get_plan().to_dict()
+            return bool(plan and plan["services"].get(self.name, {}).get("on-check-failure"))
```

Comment:
If you are going to chain `get` like that might as well start from the first one instead of checking if the dictionary is empty or not. IMO, a cleaner way would look like the following though:
```python
try:
    plan = container.get_plan().to_dict()
    return bool(plan["services"][self.name]["on-check-failure"])
except (KeyError, pebble.ConnectionError):
    return False
```
Syntax needs double checking but I believe the gist of it would work.